containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -532,6 +593,7 @@ class UnorderedMap {
         }
 
         result.set_existing(curr, free_existing);
+        if (!is_set) m_insert_op.op(m_values, curr, &v);
```

Comment:
yeah this is the only place where this is used. I think we can just have an overload of insert which takes the insert op. Or maybe even have a different name: insert_accumulate or so. Do you see any issue with that approach?

Reply:
The only issue I see would be a lot of duplicate code.